lib/SoftAP.js

```diff
@@ -186,10 +216,48 @@ SoftAP.prototype.configure = function configure(opts, cb) {
 		ch: parseInt(opts.channel)
 	};
 
+	if (opts.security & securityTable.enterprise) {
+		if (!opts.eap)
+			throw new Error('Security is set to Enterprise, but no EAP type provided');
+		if (typeof opts.eap === 'string') {
+			opts.eap = eapTypeTable[opts.eap.toLowerCase()];
+		}
+		if (opts.eap === undefined) {
+			throw new Error('Unknown EAP type provided');
+		}
+		if (opts.eap == eapTypeTable.peap) {
+			// inner identity and password are mandatory
+			opts.inner_identity = opts.inner_identity || opts.username;
+			if (!opts.inner_identity || !opts.password) {
+				throw new Error('PEAP credentials missing');
+			}
+			apConfig.ii = opts.inner_identity;
+			// Password is set later on
+		} else if (opts.eap == eapTypeTable.tls) {
+			// client certificate and private key are mandatory
+			if (!opts.private_key || !opts.client_certificate) {
+				throw new Error('EAP-TLS credentials missing');
+			}
+			apConfig.crt = opts.client_certificate.trim() + '\r\n';
+			var enc = this.aesEncrypt(opts.private_key.trim() + '\r\n');
+			apConfig.key = enc.encrypted;
+			apConfig.ek = enc.kiv;
+		}
+		apConfig.eap = opts.eap;
+		if (opts.outer_identity) {
+			apConfig.oi = opts.outer_identity;
+		}
+		opts.ca = opts.ca || opts.root_ca;
+		if (opts.ca) {
+			apConfig.ca = opts.ca.trim() + '\r\n';
```

Comment:
I know it's a small thing, but I like to factor out repeated steps (trimming and appending cr lf) so that the formatting is defined in one place only (here it appears in triplicate.) 

We could factor out into a separate function 

```
function format_certificate(cert) {
   return cert.trim() + '\r\n';
}
```

It's an optional change.